sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -430,7 +701,56 @@ def get_optimal_hyperparameters(self, name=None, namespace=None):
 
         katibexp = self.get_experiment(name, namespace=namespace)
         result = {}
-        result["currentOptimalTrial"] = katibexp.get(
-            "status", {}).get("currentOptimalTrial")
+        result["currentOptimalTrial"] = katibexp.get("status", {}).get(
+            "currentOptimalTrial"
+        )
 
         return result
+
+
+def double(min: float, max: float, step: float = None):
```

Comment:
How about gather these hyperparameter helper function to another inner module?
For example, the users could call these functions by

```python
import kubeflow.katib as katib

hp_search_space = {
    "a": katib.hp_search_space.int(min=10, max=20),
    "b": katib.hp_search_space.double(min=0.1, max=0.2)
}

katib_client.tune(
    name=name,
    objective=objective,
    hp_search_space=hp_search_space,
    objective_metric_name="result",
    max_trial_count=12
)
```

IMO, importing by `katib.int` makes me it's too board that we couldn't see it is for hyperparameter search space type immediately.

Reply:
I was even thinking to change this API in the future to support various distributions: `uniform`, `loguniform`, `categorical`, etc.

FYI, hyperopt uses [`hp` module](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/hyperopt/base_service.py#L59), skop uses [`space` module](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/base_service.py#L54), optuna uses [`distributions` module](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/optuna/base_service.py#L94), Ray uses [tune module](https://docs.ray.io/en/latest/tune/api_docs/suggestion.html?highlight=search#ax-tune-search-ax-axsearch). I think we can learn something from it. 

`search` module looks good to me, but I am not sure what is the best UX here.

What others think @terrytangyuan @gaocegege @johnugeorge @c-bata @a9p @g-votte @tenzen-y ?

Reply:
Ref https://github.com/kubeflow/katib/pull/1951#discussion_r971944632. 

IMO it feels more natural if we could support built-in lists or common numpy arrays, etc. Is there a particular motivation for bringing a separate module in the Katib SDK just to do the same thing?

Reply:
@terrytangyuan List and numpy arrays might not be enough. 
How we can identify search space for `uniform` or `loguniform` in the future?

e.g. Hyperopt utilise some numpy APIs, but it still has it's own functions for the search space: https://github.com/hyperopt/hyperopt/blob/master/hyperopt/pyll/stochastic.py#L41-L43

Reply:
The functions in that module seem to be very thin wrappers. If we could completely avoid creating additional wrappers/layers, it would be great.

Reply:
I agree with @andreyvelich.
In this PR, we can start in the minimal feature. So, we can support other distributions in other PRs.

Reply:
@terrytangyuan Do you have any ideas how we can improve UX on this ?

Reply:
I am ok as is if other projects are following the same convention.